pkg/executables/cmk.go

```diff
@@ -192,7 +195,10 @@ func (c *Cmk) ValidateZonesPresent(ctx context.Context, zones []v1alpha1.CloudSt
 func (c *Cmk) ValidateDomainPresent(ctx context.Context, domain string) (v1alpha1.CloudStackResourceIdentifier, error) {
 	domainIdentifier := v1alpha1.CloudStackResourceIdentifier{Name: domain, Id: ""}
 	command := newCmkCommand("list domains")
-	applyCmkArgs(&command, withCloudStackName(domain), appendArgs("listall=true"))
+	tokens := strings.Split(domain, domainDelimiter)
```

Comment:
Can we explain through a comment or some sort what the logic is here so we are clear that we are only looking for the last value of the domain?

Reply:
Yea agree. I made a comment stipulating a full path search and turns out the cloud monkey interface doesn't support it - would be great to have that as a inline code comment.

Reply:
Updated.